src/NLog/Targets/NetworkTarget.cs

```diff
@@ -567,6 +580,8 @@ private NetworkSender CreateNetworkSender(string address)
         {
             var sender = SenderFactory.Create(address, MaxQueueSize, OnQueueOverflow, MaxMessageSize, SslProtocols, TimeSpan.FromSeconds(KeepAliveTimeSeconds));
             sender.Initialize();
+            sender.LogEventDropped += OnLogEventDropped;
```

Comment:
Can you change so you only attach to `sender.LogEventDropped` when something is observing `NetworkTarget.LogEventDropped`?

Reply:
I think it does not make sense, as it saves few bytes on the object which is not created too often (even if you do not cache connections establishing a new one is much heavier operation than 4 bytes).

On the other had it makes the code much more complex, because if you send a log, and then attach handler it won't count dropped logs correctly anymore. You'd have to resubscribe to all cached NetworkSenders, and there are concurrency issues related to that. I think it's much better to leave it as it is.